packages/strapi-plugin-graphql/services/Loaders.js

```diff
@@ -85,71 +85,61 @@ module.exports = {
 
       // Generate constant for skip parameters.
       // Note: we shouldn't support both way of doing this kind of things in the future.
-      const skip = query.options.start || query.options.skip;
+      const skip = query.options.start || query.options.skip || 0;
+      const limit = query.options.limit || 100;
 
       // Extracting ids from original request to map with query results.
-      const ids = this.extractIds(query, ref);
-      
-      if (!_.isArray(ids)) {
-        return data
-          .filter(entry => entry[ids.alias].toString() === ids.value.toString())
-          .slice(skip, skip + query.options.limit);
-      }
+      const ids = this.extractIds(query);
+      const ast = ref.associations.find(ast => ast.alias === ids.alias);
+      const astModel = this.retrieveModel(ast.model || ast.collection, ast.plugin);
 
-      // Critical: don't touch this part until you truly understand what you're doing.
-      // The data array takes care of the sorting of the entries. It explains why we are looping from this array and not the `ids` array.
-      // Then, we're applying the `limit`, `start` and `skip` argument.
       return data
```

Comment:
Not sure of what you're doing on this part. The part was a tricky one when the order of the returned results matters. I need to test it again to make sure it works as expected...

Reply:
I think that we need to implement some unit tests for all the tricky cases so that we don't waste time on testing them manually each time, what do you think?

Reply:
Of course!! I thought that https://github.com/strapi/strapi/pull/2452#issuecomment-460048817 was exctly that. But if there are some tests missing, i can help with that.

Reply:
We aren't talking about those basic tests we are talking about implementing something more robust that covers most of the tricky cases